src/dial.js

```diff
@@ -208,7 +213,13 @@ function dial (swtch) {
     }
 
     function openConnInMuxedConn (muxer, cb) {
-      cb(muxer.newStream())
+      muxer.newStream((err, conn) => {
+        if (err) {
+          return cb(err)
+        }
+
+        cb(null, conn)
+      })
```

Comment:
👍

Reply:
why not just newStream(cb)?

Reply:
Good question - I'm not sure if this is the correct behavior either, this change is related to -https://github.com/libp2p/js-libp2p-mplex/pull/75, which would simply `throw` on error and crash the whole process. Since I return the error now, it seemed logical to handle it here, but I wasn't sure it is the correct approach, specially since other multiplexers (spdy) don't seem to propagate the errors in the same way. 

So the question remains - should we handle the errors in `switch` or just ignore it?